packages/seacas/libraries/ioss/src/main/io_shell.C

```diff
@@ -80,10 +81,12 @@ int main(int argc, char *argv[])
 #ifdef SEACAS_HAVE_MPI
   MPI_Init(&argc, &argv);
   MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+  ON_BLOCK_EXIT(MPI_Finalize);
 #endif
 
 #ifdef SEACAS_HAVE_KOKKOS
   Kokkos::initialize(argc, argv);
```

Comment:
Literally you could just use Kokkos::ScopeGuard, like this:
```c++
Kokkos::ScopeGuard kokkosScope (argc, argv);
```
Its destructor calls Kokkos::finalize.